products.js

```diff
@@ -0,0 +1,308 @@
+          expectedText = expectedDom.body.firstChild.src;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.src)[0];
+        } else if (this.feature === 'title') {
+          //compare innerHTML text of titles
+          expectedText = expectedDom.head.firstChild.innerHTML;
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.innerHTML)[0];
+        } else if (this.feature === 'price') {
+          //strip whitespace and dollar sign if there is one when comparing price
+          expectedText = expectedDom.body.firstChild.textContent.replace('$', '').trim();
+          gotText = this.contentFnodes(sourceDom).map(fnode => fnode.element.textContent.replace('$', '').trim())[0];
+        }
+
+        this.lengthOfExpectedTexts++;
+        if(expectedText !== gotText) {
+          this.lengthOfDiffs++;
+        }
+
+        // Uncomment for debugging:
+        console.log(leven(expectedText, gotText), expectedText.length, leven(expectedText, gotText)/expectedText.length);
+        console.log('Got:\n' + gotText);
+        console.log('\nExpected:\n' + expectedText);
+    }
+
+    score() {
+        return this.lengthOfDiffs / this.lengthOfExpectedTexts * 100;
+    }
+}
+
+function createDict(item, sourceDom){
+  var nodesDict = JSON.parse(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', item, 'nodes.txt'), 'utf-8'));
+  var elems = sourceDom.getElementsByTagName('*');
+  console.log(sourceDom.getElementsByTagName('*').length, Object.keys(nodesDict).length);
+
+  //Match each node from the sourceDom to its css, and store in the global dict
+  for(let i = 0; i < Object.keys(nodesDict).length; i++){
+    currDict[elems[i]] = nodesDict[i.toString()];
+  }
+}
+
+
+function deviationScore(folders, feature, coeffs = []) {
+    var stats;
+    if(feature === 'image') {
+      stats = new DiffStats(tunedImageFnodes(...coeffs), feature);
+    } else if (feature === 'title') {
+      stats = new DiffStats(tunedTitleFnodes(...coeffs), feature);
+    } else {
+      stats = new DiffStats(tunedPriceFnodes(...coeffs), feature);
+    }
+
+    //For each test file, create the object-> css map, and run the comparison function
+    folders.forEach(function(item){
+      if(['macys', 'swatch'].includes(item)){return;} //random jsdom script tag parsing errors, ignore for now
+      const domFromFile = fileName => staticDom(readFileSync(join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', item, fileName)));
+
+      createDict(item, domFromFile('source.html'));
+      var pair = [domFromFile('expected-' + feature + '.html'), domFromFile('source.html')];
+      stats.compare(...pair);
+      console.log(stats.score());
+      currDict = {};
+    });
+
+    return stats.score();
+}
+
+if (require.main === module) {
+    const features = ['title', 'image', 'price'];
+    const dirs = p => readdirSync(p).filter(f => statSync(p+"/"+f).isDirectory());
```

Comment:
Some nitpickers would complain about the lack of conventional spaces around binary operators like `+` but not me—heavens, no. ;-)